.github/workflows/main.yml

```
@@ -0,0 +1,55 @@
+name: CI
+
+on:
+  push:
+    branches: ['master']
+  pull_request:
+    branches: ['*']
+
+jobs:
+  build_and_test:
+    name: LNDK Rust Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+      - uses: actions-rs/cargo@v1
+        name: cargo build
+        with:
+          command: build
+          args: --release --all-features
+      - uses: actions-rs/cargo@v1
+        name: cargo test
+        with:
+          command: test
+          args: --all-targets --benches --frozen
+      - uses: actions-rs/cargo@v1
+        name: cargo fmt
+        with:
+          command: fmt
+          args: --check
+      - uses: actions-rs/cargo@v1
+        name: cargo clippy
+        with:
+          command: clippy
+          args: -- --deny warnings
+  coverage:
+    name: LNDK Code Coverage
+    runs-on: ubuntu-latest
+    env:
+      CARGO_TERM_COLOR: always
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install Rust
+        run: rustup update stable
```

This seems to break since `rustup` isn't available here. So we can just use the `actions-rs/toolchain@v1` action again with `stable` toolchain.

```suggestion
      - uses: actions-rs/toolchain@v1
        with:
          toolchain: stable
```

---

This step [succeeds in my fork](https://github.com/thinkmassive/lndk/actions/runs/4387096599/jobs/7681971751), so I'm not sure I follow. It also looks like the [rust-cache](https://github.com/marketplace/actions/rust-cache#example-usage) action suggests running `rustup` using `actions/checkout@v3`.

In any case, this suggestion appears to [work well](https://github.com/thinkmassive/lndk/actions/runs/4408946983/jobs/7724614619), so I trust that it's preferable to running rust commands in the `checkout` action.

---

Hmm. Might just be a limitation of `act` then! My GitHub Actions knowledge is somewhat limited. Anyway, think we can tweak it later :)

---

Sounds good to me! Maybe your local Actions are using a slightly different Runner. Here are the rust tools that should be available in the `ubuntu-latest` hosted runner:
https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#rust-tools